Sources/NIOHPACK/HeaderTables.swift

```diff
@@ -0,0 +1,58 @@
+// swift-tools-version:4.0
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+struct HeaderTableEntry {
+    var name: String
+    var value: String?
```

Comment:
I'd seen the HTTP/1 headers implementation and thought it was designed to avoid parsing out header blocks from their wire format; that didn't gel with HTTP/2 though, since there's an extra decoding step required, so I'd dismissed it. I can kind of see how this would work though, along with making the header tables into raw ByteBuffer instances---it seems that the HTTP/1 HTTPHeaders implementation would work pretty well for the header tables themselves, in fact, as you suggest later on.

Reply:
> I'd seen the HTTP/1 headers implementation and thought it was designed to avoid parsing out header blocks from their wire format

That's it's primary purpose, but it has the beneficial side effect of avoiding the strings as well. 👍